src/block_cache/block_chunkcache.c

```diff
@@ -164,23 +171,55 @@ __chunkcache_free_chunk(WT_SESSION_IMPL *session, WT_CHUNKCACHE_CHUNK *chunk)
 }
 
 /*
- * __chunkcache_make_hash --
- *     Populate the hash data structure, which uniquely identifies the chunk.
+ * __chunkcache_tmp_hash --
+ *     Populate the hash data structure, which uniquely identifies the chunk. The hash ID we
+ *     populate will contain a pointer to the block name, thus the block name must outlive the hash
+ *     ID.
  */
 static inline uint64_t
-__chunkcache_make_hash(WT_CHUNKCACHE *chunkcache, WT_CHUNKCACHE_HASHID *hash_id, WT_BLOCK *block,
+__chunkcache_tmp_hash(WT_CHUNKCACHE *chunkcache, WT_CHUNKCACHE_HASHID *hash_id, WT_BLOCK *block,
   uint32_t objectid, wt_off_t offset)
 {
-    uint64_t hash;
+    WT_CHUNKCACHE_INTERMEDIATE_HASH intermediate;
+    uint64_t hash_final;
+
+    WT_CLEAR(intermediate);
+    intermediate.name_hash = __wt_hash_city64(block->name, strlen(block->name));
+    intermediate.objectid = objectid;
+    intermediate.offset = WT_CHUNK_OFFSET(chunkcache, offset);
 
+    /*
+     * The hashing situation is a little complex. We want to construct hashes as we iterate over the
+     * chunks we add/remove, and these hashes consist of an object name, object ID, and offset. But
+     * to hash these, the bytes need to be contiguous in memory. Having the object name as a
+     * fixed-size character array would work, but it would need to be large, and that would waste a
+     * lot of space most of the time. The alternative would be to allocate a new structure just for
+     * hashing purposes, but then we're allocating/freeing on the hot path.
+     *
+     * Instead, we hash the object name separately, then bundle that hash into a temporary (stack
+     * allocated) structure with the object ID and offset. Then, we hash that intermediate
+     * structure.
+     */
     WT_CLEAR(*hash_id);
     hash_id->objectid = objectid;
-    memcpy(&hash_id->objectname, block->name, WT_MIN(strlen(block->name), WT_CHUNKCACHE_NAMEMAX));
     hash_id->offset = WT_CHUNK_OFFSET(chunkcache, offset);
-    hash = __wt_hash_city64((void *)hash_id, sizeof(WT_CHUNKCACHE_HASHID));
+    hash_id->objectname = block->name;
+
```

Comment:
Just thinking ahead when the chunk cache will be persistent. In that scenario, copying a pointer to object name won't work, because the chunk metadata (a persistent piece of information) would outlive the block data structure (a volatile piece of information).